src/System.Diagnostics.Process/src/System/Diagnostics/Process.Linux.cs

```diff
@@ -57,6 +58,40 @@ internal DateTime StartTimeCore
             }
         }
 
+        /// <summary>Computes a time based on a number of ticks since boot.</summary>
+        /// <param name="timespanAfterBoot">The timespan since boot.</param>
+        /// <returns>The converted time.</returns>
+        internal static DateTime BootTimeToDateTime(TimeSpan timespanAfterBoot)
+        {
+            // Use the uptime and the current time to determine the absolute boot time.
+            DateTime bootTime = DateTime.UtcNow - Uptime;
+
+            // And use that to determine the absolute time for timespan.
+            DateTime dt = bootTime + timespanAfterBoot;
+
+            // The return value is expected to be in the local time zone.
+            // It is converted here (rather than starting with DateTime.Now) to avoid DST issues.
+            return dt.ToLocalTime();
+        }
+
+        /// <summary>Gets the elapsed time since the system was booted.</summary>
+        private static TimeSpan Uptime
+        {
+            get
+            {
+                // '/proc/uptime' accounts time a device spends in sleep mode.
+                const string uptimeFile = "/proc/uptime";
+                string text = File.ReadAllText(uptimeFile);
+
+                double uptimeSeconds;
+                int length = text.IndexOf(' ');
+                bool success = Double.TryParse(text.Substring(0, length), NumberStyles.AllowDecimalPoint, NumberFormatInfo.InvariantInfo, out uptimeSeconds);
```

Comment:
You can use `text.AsReadOnlySpan().Slice(0, length)` to avoid the substring allocation.

Reply:
nice :sunglasses: